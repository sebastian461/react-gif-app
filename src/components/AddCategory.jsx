import { useState } from "react";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    console.log(inputValue);

    if (inputValue.trim().length < 1) return;

    /* Para evitar enviar las categorías como una prop
    al usar el método generado con el 'useState' se pueden usar las categorias por un callback */
    setCategories((categories) => [inputValue, ...categories]);
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit}>
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};
