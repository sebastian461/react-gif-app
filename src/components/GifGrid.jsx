import { GifItem } from "./GifItem";

import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {
  /* Todo este código se puede resumir como un hook personalizado */
  /* const [images, setImages] = useState([]);

  const getImages = async () => {
    const newImages = await getGifs(category);
    setImages(newImages);
  };

  //Es mala practica usar funciones directamente en un componente, por eso se usa el hook 'useEffect'
  //se le indica al componente 
  useEffect(() => {
    getImages();
  }, []); //si el arreglo está vació, el componente ejecuta el callback solo la primera vez que se renderiza */

  const { images, isLoading } = useFetchGifs(category);
  console.log({ isLoading });

  return (
    <>
      <h3>{category}</h3>
      {/* Para agregar clases se usa la palabra 'className' */}
      <div className="card-grid">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};
