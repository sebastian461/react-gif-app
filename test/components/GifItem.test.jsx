import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas en GifItem", () => {
  const titleTest = "Titulo de prueba";
  const urlTest = "http://urltest.com/dragonball.jpg";

  test("snapshot debe regresar un error al no enviar titulo y url", () => {
    const { container } = render(<GifItem title={titleTest} url={urlTest} />);
    //const { container } = render(<GifItem />);
    expect(container).toMatchSnapshot();
  });

  test("debe mostrar la imagen con el URL y el ALT indicado", () => {
    render(<GifItem title={titleTest} url={urlTest} />);
    //expect(screen.getByRole("img").src).toBe(urlTest);
    //expect(screen.getByRole("img").alt).toBe(titleTest);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(urlTest);
    expect(alt).toBe(titleTest);
  });

  test("debe mostrar el titulo en el componente", () => {
    render(<GifItem title={titleTest} url={urlTest} />);
    expect(screen.getByText(titleTest)).toBeTruthy();
  });
});
