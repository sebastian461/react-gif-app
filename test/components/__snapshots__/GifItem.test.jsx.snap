// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem snapshot debe regresar un error al no enviar titulo y url 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo de prueba"
      src="http://urltest.com/dragonball.jpg"
    />
    <p>
      Titulo de prueba
    </p>
  </div>
</div>
`;
